Sources/Core/X86/SoftwareBreakpointManager.cpp

```diff
@@ -51,10 +51,7 @@ int SoftwareBreakpointManager::hit(Target::Thread *thread, Site &site) {
 
 void SoftwareBreakpointManager::getOpcode(uint32_t type,
                                           ByteVector &opcode) const {
-  DS2ASSERT(type == 1);
```

Comment:
I might be missing something, but I think this assertion should stay here. If everything goes as expected, either we were asked to set a breakpoint of size 1, or we were sent 0, and we chose a breakpoint of size 1 in `chooseBreakpointSize()`, or we were sent some other value, and we should have returned an error to the debugger and *not* added the breakpoint. So here, type should *always* be `1`.

Reply:
True.